### ISSUE TITLE
Gmail sync never completes...

### ISSUE THREAD

**Issue by dzg** [imapsync/imapsync#171]
I'm trying to copy all messages from my personal @gmail.com to my G Suite account:

```
imapsync \
--user1 ___@gmail.com --password1 ___ --user2 ___@mydomain.com --password2 ___ \
--no-modulesversion --gmail1 --gmail2 --folder "[Gmail]/All Mail" --nofoldersizes \
--useheader "MESSAGE-ID" --addheader \
--usecache --tmpdir ~/LOG_imapsync/ --noresyncflags --debugflags
```

First run copied several thousand messages.
Second run copied another several thousand.

Shouldn't it be done in one run? Do I have to keep running it?

I also tried adding `--subfolder2 imapsync` to label synced messages on account2, but that had no effect.

Is there any way to add a label on account1 to each message copied so that subsequent runs can ignore those?

Note, even running in batches (with `--maxage 300`) keeps copying more messages on every run.

Also ... given a line in the log such as:

    msg [Gmail]/All Mail/160735 {10824465}   copied to [Gmail]/All Mail/216771     0.02 msgs/s  42.236 KiB/s 3.069 GiB copied

Is there any way to find "message 160735" on account1 from within Gmail, say, via URL or search?

If I use `--debug` I can see the ID of each messsage but then the ouput is enormous.

**Comment by dzg** [comment]
Also ... given a line in the log such as:

    msg [Gmail]/All Mail/160735 {10824465}   copied to [Gmail]/All Mail/216771     0.02 msgs/s  42.236 KiB/s 3.069 GiB copied

Is there any way to find "message 160735" on account1 from within Gmail, say, via URL or search?

If I use `--debug` I can see the ID of each messsage but then the ouput is enormous.

**Comment by dzg** [comment]
Note, even running in batches (with `--maxage 300`) keeps copying more messages on every run.

**Comment by greem** [comment]
Have you followed the details here?

https://imapsync.lamiral.info/FAQ.d/FAQ.Gmail.txt

"Q. How to synchronize from Gmail to Gmail?"

**Comment by dzg** [comment]
> Have you followed the details here?
> https://imapsync.lamiral.info/FAQ.d/FAQ.Gmail.txt
> "Q. How to synchronize from Gmail to Gmail"?

Yes I have, several times.

**Comment by gilleslamiral** [comment]
Can you try a resync without --nofoldersizes nor --usecache

--usecache is not recommended with Gmail, it's useless or counterproductive. 

https://imapsync.lamiral.info/FAQ.d/FAQ.Gmail.txt
...
````
Q. Can I safely use --useuid or --usecache for Gmail transfers?

R. Yes, but I suggest to not use --useuid for Gmail transfers.

Using UIDs is useless with Gmail in the case of global duplicates
(duplicates across different folders). Gmail always accept a global
duplicate message as a new message, giving imapsync a new UID for this
message, and throw it away because it already has it. Gmail
will do this at each run so imapsync will always try to copy the
message, and Gmail will always accept and throw away the new copy. It
ends up with no duplicates on Gmail but a waste of bandwith and time,
which is the opposite goal of --usecache implied by --useuid.
````

**Comment by dzg** [comment]
> Can you try a resync without --nofoldersizes nor --usecache

Yes I will try.

But why no `--nofoldersizes`? That adds a lot of time to the sync. Is it important?

Merci!

**Comment by gilleslamiral** [comment]
It is not essential to the sync itself but I suggest to not use  `--nofoldersizes` because it helps so much that the supplementary time is usually less than the time spending to understand what happened, what went wrong, what went good, how long will it take to finish the sync etc.

**Comment by gilleslamiral** [comment]
To sum up:

1) Don't use --usecache nor --useuid when syncing to Gmail. 

2) Since this issue was posted I've coded ` --synclabels ` and ` --resynclabels ` options. They are activated by default when using ` --gmail1 --gmail2 `. It does a tremendous bandwidth gain with Gmailboxes that have massive labels usage or when used with ` --subfolder2 ` or ` --subfolder1 `. I took care to avoid a "never ending sync" syndrome, ie, imapsync resyncing messages at each run while nothing changed on host1. Believe me, it was a pain in the ass.

Closing.
